platforms/software/testing-base-infrastructure/src/main/java/org/gradle/api/internal/tasks/testing/TestResultProcessor.java

```diff
@@ -43,6 +44,12 @@ public interface TestResultProcessor {
     @UsedByScanPlugin("test-distribution")
     void output(Object testId, TestOutputEvent event);
 
+    /**
+     * Notifies this processor that a test has produced some metadata.
+     */
+    @UsedByScanPlugin("test-distribution")
```

Comment:
I removed the event as it isn't needed yet, but I think this should stay in the public package, as that seems to be where it will go when it is needed (that's where the `TestOutputEvent` lives).